WearVerifyRemoteApp/Wearable/src/main/java/com/example/android/wearable/wear/wearverifyremoteapp/MainWearActivity.kt

```diff
@@ -0,0 +1,206 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.example.android.wearable.wear.wearverifyremoteapp
+
+import android.content.Intent
+import android.net.Uri
+import android.os.Bundle
+import android.util.Log
+import androidx.core.view.doOnPreDraw
+import androidx.core.view.isInvisible
+import androidx.core.view.isVisible
+import androidx.core.view.setPadding
+import androidx.fragment.app.FragmentActivity
+import androidx.lifecycle.lifecycleScope
+import androidx.wear.phone.interactions.PhoneTypeHelper
+import androidx.wear.remote.interactions.RemoteIntentHelper
+import androidx.wear.widget.ConfirmationOverlay
+import com.example.android.wearable.wear.wearverifyremoteapp.databinding.ActivityMainBinding
+import com.google.android.gms.wearable.CapabilityClient
+import com.google.android.gms.wearable.CapabilityInfo
+import com.google.android.gms.wearable.Node
+import com.google.android.gms.wearable.NodeClient
+import com.google.android.gms.wearable.Wearable
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.guava.await
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.tasks.await
+import kotlinx.coroutines.withContext
+import kotlin.math.roundToInt
+import kotlin.math.sqrt
+
+/**
+ * Checks if the phone app is installed on remote device. If it is not, allows user to open app
+ * listing on the phone's Play or App Store.
+ */
+class MainWearActivity : FragmentActivity(), CapabilityClient.OnCapabilityChangedListener {
+
+    private lateinit var binding: ActivityMainBinding
+
+    private lateinit var capabilityClient: CapabilityClient
+    private lateinit var nodeClient: NodeClient
+    private lateinit var remoteIntentHelper: RemoteIntentHelper
+
+    private var androidPhoneNodeWithApp: Node? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        Log.d(TAG, "onCreate()")
+        super.onCreate(savedInstanceState)
+
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        capabilityClient = Wearable.getCapabilityClient(this)
+        nodeClient = Wearable.getNodeClient(this)
+        remoteIntentHelper = RemoteIntentHelper(this)
+
+        binding.informationTextView.text = getString(R.string.message_checking)
+        binding.remoteOpenButton.setOnClickListener { 
+            openAppInStoreOnPhone()
+        }
+
+        if (resources.configuration.isScreenRound) {
+            binding.scrollingContentContainer.doOnPreDraw {
```

Comment:
doOnNextLayout?

Reply:
`doOnLayout` and `doOnNextLayout` let the default/0 padding be drawn for one frame, which `doOnPreDraw` avoids.

Reply:
Fascinating, and unexpected considering that padding can affect layout, and layout should happen before draw...

Reply:
Just dived into the source for this more, I was curious what the exact difference was.

You are absolutely correct that layout will happen before draw, but `doOnPreDraw` has the special behavior of [cancelling the draw call](https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/java/android/view/ViewTreeObserver.java;l=1085-1100) for the layout pass without the padding, so the user only sees the layout with the padding.

The most ideal solution to this would probably be a subclass of the layout that does all of this in `onMeasure` to avoid the double layout pass (a la `BoxInsetLayout`), but that seems like a bit of overkill here.